### PR TITLE
Bump Elixir to v1.6.1

### PR DESCRIPTION
**Pull request by maukoquiroga**
In order to use ```ExUnit.Callbacks.start_supervised!```, for testing ```GenServer```.

### CODE REVIEW

File: .circleci/Dockerfile [hunk 1]
```diff
@@ -28,8 +28,8 @@ RUN curl https://gist.githubusercontent.com/maukoquiroga/48e9aa501ca8c7d63615644
 RUN chmod +x /etc/init.d/phantomjs
 
 #Install data-validator
-RUN apt-get install -y mongodb-server python-pip python-virtualenv
-RUN git clone https://github.com/etalab/transport-validator
+RUN apt-get install -y mongodb-server python-dev python-pip python-virtualenv
+RUN git clone https://github.com/maukoquiroga/transport-validator -b use-ipython-5
```
Comment (l-vincent-l) [inline .circleci/Dockerfile]: Change it to the main repo

File: lib/transport_web/api/views/dataset_view.ex [hunk 2]
```diff
@@ -6,6 +6,9 @@ defmodule TransportWeb.API.DatasetView do
   end
 
   def render(_conn, %{errors: errors}) do
-    Poison.encode(errors)
+    case Poison.encode(errors) do
```
Comment (l-vincent-l) [inline lib/transport_web/api/views/dataset_view.ex]: It changes quite a bit the return of this function, was it a bug before?

Comment (maukoquiroga) [reply lib/transport_web/api/views/dataset_view.ex]: It is weird as it was not failing, but it should have! I say it was a bug before.

File: .circleci/Dockerfile (hunk 1 above)
Comment (maukoquiroga) [reply .circleci/Dockerfile]: Done

**Pull request merged**